bin/accessibility-automation/helper.js

```diff
@@ -0,0 +1,193 @@
+const { API_URL } = require('./constants');
+const utils = require('../helpers/utils');
+const fs = require('fs');
+const path = require('path');
+const request = require('request');
+const os = require('os');
+const glob = require('glob');
+const helper = require('../helpers/helper');
+
+exports.checkAccessibilityPlatform = (user_config) => {
+  let accessibility = false;
+  try {
+    user_config.browsers.forEach(browser => {
+      if (browser.accessibility) {
+        accessibility = true;
+        return true;
+      }
+    })
+  } catch {}
+
+  return accessibility;
+}
+
+exports.setAccessibilityCypressCapabilities = async (user_config, accessibilityResponse) => {
+  if (utils.isUndefined(user_config.run_settings.accessibilityOptions)) {
+    user_config.run_settings.accessibilityOptions = {}
+  }
+  user_config.run_settings.accessibilityOptions["authToken"] = accessibilityResponse.data.accessibilityToken;
+  user_config.run_settings.accessibilityOptions["auth"] = accessibilityResponse.data.accessibilityToken;
+  user_config.run_settings.accessibilityOptions["scannerVersion"] = accessibilityResponse.data.scannerVersion;
+  user_config.run_settings.system_env_vars.push(`ACCESSIBILITY_AUTH=${accessibilityResponse.data.accessibilityToken}`)
+  user_config.run_settings.system_env_vars.push(`ACCESSIBILITY_SCANNERVERSION=${accessibilityResponse.data.scannerVersion}`)
+}
+
+exports.createAccessibilityTestRun = async (user_config, framework) => {
+
+  try {
+    const userName = user_config["auth"]["username"];
+    const accessKey = user_config["auth"]["access_key"];
+    let settings = utils.isUndefined(user_config.run_settings.accessibilityOptions) ? {} : user_config.run_settings.accessibilityOptions
+
+    const {
+      buildName,
+      projectName,
+      buildDescription
+    } = helper.getBuildDetails(user_config);
+
+    const data = {
+      'projectName': projectName,
+      'buildName': buildName,
+      'startTime': (new Date()).toISOString(),
+      'description': buildDescription,
+      'source': {
+        frameworkName: "Cypress",
+        frameworkVersion: helper.getPackageVersion('cypress', user_config),
+        sdkVersion: helper.getAgentVersion()
+      },
+      'settings': settings,
+      'versionControl': await helper.getGitMetaData(),
+      'ciInfo':  helper.getCiInfo(),
+      'hostInfo': {
+        hostname: os.hostname(),
+        platform: os.platform(),
+        type: os.type(),
+        version: os.version(),
+        arch: os.arch()
+      },
+      'browserstackAutomation': process.env.BROWSERSTACK_AUTOMATION === 'true'
+    };
+
+    const config = {
+      auth: {
+        user: userName,
+        pass: accessKey
+      },
+      headers: {
+        'Content-Type': 'application/json'
+      }
+    };
+
+    const response = await nodeRequest(
+      'POST', 'test_runs', data, config, API_URL
+    );
+    process.env.BS_A11Y_JWT = response?.data?.data?.accessibilityToken;
```

Comment:
There are users who use node 12 for running cypress CLI. Optional chaining would fail for them. Better use null checks rather than optional chaining.

Reply:
done